src/lobby.cpp

```diff
@@ -333,12 +336,11 @@ void Lobby::on_add_to_fav_pressed()
 void Lobby::on_add_to_fav_released()
 {
   ui_add_to_fav->set_image("addtofav");
-
-  // you cant add favorites from favorites m8
-  if (!public_servers_selected)
-    return;
-
-  ao_app->add_favorite_server(get_selected_server());
+  if (public_servers_selected) {
+    if (int selection = get_selected_server(); selection > -1) {
```

Comment:
Put declarations on a different line than the if-statement if the comparison is more than a simple null check, please.

Reply:
It's a new feature part of C++17 https://en.cppreference.com/w/cpp/language/if

I don't mind moving out the declaration, just want one more comment as confirmation. I assume for readability/style.

Reply:
Yes, it's for readability.